subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/TransformInfo.java

```diff
@@ -51,23 +50,12 @@
     protected List<File> result;
     protected Throwable failure;
 
-    public static TransformInfo from(
-        List<UserCodeBackedTransformer> transformerChain,
-        BuildableSingleResolvedArtifactSet artifact
-    ) {
-        Iterator<UserCodeBackedTransformer> iterator = transformerChain.iterator();
-        TransformInfo current;
-        TransformInfo previous = null;
-        do {
-            UserCodeBackedTransformer transformer = iterator.next();
-            if (previous == null) {
-                current = new InitialTransformInfo(transformer, artifact);
-            } else {
-                current = new ChainedTransformInfo(transformer, previous);
-            }
-            previous = current;
-        } while (iterator.hasNext());
-        return current;
+    public static TransformInfo chained(List<UserCodeBackedTransformer> transformerChain, TransformInfo previous) {
+        return new ChainedTransformInfo(transformerChain.get(transformerChain.size() - 1), previous);
+    }
```

Comment:
This should not receive a `List` only to get the last element of it...

Reply:
I am now passing the last element of the list.